### ISSUE TITLE
Seneca don't respect default$ parambulator rule.

### ISSUE THREAD

**Issue by StarpTech** [senecajs/seneca#395]
Seneca: 2.0.0

Issue 1:
Define an action
```
    level: {
      type$: 'number',
      default$: 0
    },
```

Call an act and get this as result:
```
....
  'default$': undefined,
```
 `level` is not prestent in the message (undefined).

Issue 2:

Define an action
```
    level: {
      type$: 'number',
      required: true,
      default$: 0
    },
```
Now I can send an empty payload and it will be validated as valid.
```
{}
```

**Comment by mcdonnelldean** [comment]
I don't see anything obvious in the code to see why this isn't working? What code are you running? I want to maybe add some tests to prove this out.

**Comment by StarpTech** [comment]
@mcdonnelldean full example:

```js
var seneca = require('seneca')()

seneca.add({ cmd: 'salestax', level: { default$: 1, type$: 'number' } }, function (args, callback) {
  callback(null, { level: args.level })
})

seneca.act({ cmd: 'salestax' }, function (err, result) {
  console.log(result) //{ level: undefined }
})
```

**Comment by mcdonnelldean** [comment]
@StarpTech What are you getting from the error object?

**Comment by mcdonnelldean** [comment]
Nevermind, I just ran, it looks like it's chewing it up and not reporting errors. I've not used this stuff before so it will take a few days to drill down. If @geek is free he may be able to take a look too. I'll leave it as investigate for now and mark as a bug as soon as I get speaking to someone who knows their way around parambulator.

**Comment by StarpTech** [comment]
@mcdonnelldean @geek I tried this sample with the parambulator (1.5.1) validator and it worked. You can mark it as bug.

**Comment by StarpTech** [comment]
@geek @mcdonelldean any updates?

**Comment by mcdonnelldean** [comment]
@StarpTech No. Currently I'm working on the TCP reconnect issue, there is a bug in the new implementation. @geek Is away for a while so not working. If you can add a test case to the suite it would speed things up. As it stands TCP wins in priority.

**Comment by StarpTech** [comment]
@mcdonnelldean thanks I will try to write a test.

**Comment by mcdonnelldean** [comment]
Fixed via #422. Thanks @StarpTech

**Comment by mcdonnelldean** [comment]
Monday. I have a couple of things that need to go out with this but they won't be baked until tonight. It may go out sooner but Monday is the latest.
